### PR TITLE
Jordan/1173 addresses

### LINKED ISSUE #1173: Cosmos address component
the UI can abbreviate addresses and pubkeys for a friendlier and less cluttered UI. 

though, bech32 prefixes have made our address _a bit_ more readable - they are still for computers (let's be honest).

i'd like to propose that we abbreviate addresses and pubkeys - the tooltip could display the full address. see below.

this will result in a more readable, less overwhelming, cleaner UI. 🥇 
```
cosmosvalpub12ztjt9nfj20cxw4w4fdg5g0lak83mc23csqzgx
cosmosvalpub...qzgx

cosmosaccaddr1qk2q3l28t6qhzyc7psdffsdf832ep8uwdaer
cosmosaccaddr...daer
```

### LINKED ISSUE #1437: Refactor ShortAddress
Currently the component handles other formats that are not bech32. I don't hink we need this and it also hides wrong addresses.
Additionally we want to rename ShortAddress to ShortBech32.

### PR DESCRIPTION
**Pull request by jbibla**
Closes #1173 
Closes #1437 

Description:
- new `ShortBech32` component that includes the short address script, copy ux, and a tooltip
- lots of updates to address format in tests
- improved `TmBalance` css 

❤️ Thank you!

### CODE REVIEW

File: test/unit/specs/components/staking/__snapshots__/LiValidator.spec.js.snap [hunk 1]
```diff
@@ -26,12 +26,29 @@ exports[`LiValidator has the expected html structure 1`] = `
         <div
           class="top"
         >
-          mr_mounty
+          herr_schmidt_revoked
```
Comment (jbibla) [inline test/unit/specs/components/staking/__snapshots__/LiValidator.spec.js.snap]: 🐱

File: app/src/renderer/components/common/ShortAddress.vue [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+<script>
+import { clipboard } from "electron"
+export default {
+  name: `short-address`,
+  props: [`address`],
+  data: () => ({
+    showSuccess: false
+  }),
+  computed: {
+    shortAddress({ address } = this, length = 4) {
```
Comment (fedekunze) [inline app/src/renderer/components/common/ShortAddress.vue]: question: do we always want to show the last 4 chars ? maybe getting the `length` value from the props is a better option ?

File: app/src/renderer/components/common/ShortAddress.vue [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+  .address(@click.prevent.stop="copy" v-tooltip.top="address")#address {{ shortAddress }}
+  .success(:class="{active:showSuccess}")
+    i.material-icons check
+    span Copied
+</template>
+
+<script>
+import { clipboard } from "electron"
+export default {
+  name: `short-address`,
```
Comment (fedekunze) [inline app/src/renderer/components/common/ShortAddress.vue]: I'd rename the component to `short-bech32` since we are using it for addresses and pubkeys

File: app/src/renderer/components/common/ShortAddress.vue (hunk 2 above)
Comment (jbibla) [reply app/src/renderer/components/common/ShortAddress.vue]: right now, we are going to show only the last 4 - but it will be easy to change it if we decide we should.

File: app/src/renderer/components/common/ShortAddress.vue (hunk 3 above)
Comment (jbibla) [reply app/src/renderer/components/common/ShortAddress.vue]: i think this is a good suggestion!

Comment (fedekunze) [reply app/src/renderer/components/common/ShortAddress.vue]: we can change the name later tho, nbd

File: test/unit/specs/components/staking/LiValidator.spec.js [hunk 4]
```diff
@@ -2,32 +2,50 @@ import setup from "../../../helpers/vuex-setup"
...
 
-  beforeEach(async () => {
-    let test = instance.mount(LiValidator, {
+  beforeEach(() => {
+    let instance = mount(LiValidator, {
       propsData: {
         validator: {
           id: `abc`,
-          description: {}
+          pub_key: `valpub123456789`,
```
Comment (fedekunze) [inline test/unit/specs/components/staking/LiValidator.spec.js]: you can use the lcd mocked validators instead 👍

File: app/src/renderer/components/common/ShortAddress.vue (hunk 3 above)
Comment (faboweb) [reply app/src/renderer/components/common/ShortAddress.vue]: Good suggestion.

Comment (NodeGuy) [reply app/src/renderer/components/common/ShortAddress.vue]: Yes, I agree.

Comment (jbibla) [reply app/src/renderer/components/common/ShortAddress.vue]: agree that i should rename it or agree that it can wait until later? (or both)

Comment (fedekunze) [reply app/src/renderer/components/common/ShortAddress.vue]: I think they were talking about renaming the component

Comment (faboweb) [reply app/src/renderer/components/common/ShortAddress.vue]: I think we should rename it. We can so later.

File: app/src/renderer/components/common/ShortAddress.vue [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+  props: [`address`],
+  data: () => ({
+    showSuccess: false
+  }),
+  computed: {
+    shortAddress({ address } = this, length = 4) {
+      if (!address) return `Address Not Found`
+      if (address.indexOf(`1`) === -1) {
```
Comment (faboweb) [inline app/src/renderer/components/common/ShortAddress.vue]: How can a bech32 address not have a 1 in it?

File: app/src/renderer/components/wallet/PageWallet.vue [hunk 6]
```diff
@@ -14,8 +14,7 @@ tm-page(data-title="Wallet", :title="config.devMode ? '' : 'Wallet'")
     tm-list-item(
       :title="wallet.address"
       :btn="'Receive'"
-      :overflow="true"
-      @click.native="copy")
```
Comment (faboweb) [inline app/src/renderer/components/wallet/PageWallet.vue]: Why was this removed?

File: test/unit/specs/components/common/ShortAddress.spec.js [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+    wrapper.update()
+    expect(wrapper.vm.shortAddress).toBe(`add1asd`)
+
+    wrapper.setProps({ address: `add2asd` })
+    wrapper.update()
+    expect(wrapper.vm.shortAddress).toBe(`add2asd`)
+  })
+
+  it(`should return a short address with the first 4 letters`, () => {
+    wrapper.setProps({ address: `cosmosaddress2asdfasdfasdf` })
```
Comment (faboweb) [inline test/unit/specs/components/common/ShortAddress.spec.js]: shouldn't this fail as it is not bech32?

File: test/unit/specs/components/common/ShortAddress.spec.js [hunk 8]
```diff
@@ -0,0 +1,73 @@
...
+        .includes(`active`)
+    ).toBe(false)
+    wrapper.find(`.address`).trigger(`click`)
+    expect(
+      wrapper
+        .find(`.success`)
+        .classes()
+        .includes(`active`)
+    ).toBe(true)
+    await sleep(3500)
```
Comment (faboweb) [inline test/unit/specs/components/common/ShortAddress.spec.js]: please do not use sleep, use `jest.useFakeTimers()`

File: test/unit/specs/components/common/TmBalance.spec.js [hunk 9]
```diff
@@ -4,10 +4,10 @@ import TmBalance from "common/TmBalance"
 
 describe(`TmBalance`, () => {
   let wrapper
-  let { mount } = setup()
+  let { shallow } = setup()
 
   beforeEach(() => {
-    let instance = mount(TmBalance, {
+    let instance = shallow(TmBalance, {
```
Comment (faboweb) [inline test/unit/specs/components/common/TmBalance.spec.js]: why was this changed?

File: test/unit/specs/components/staking/__snapshots__/TabValidators.spec.js.snap [hunk 10]
```diff
@@ -89,21 +94,26 @@ exports[`PageStaking has the expected html structure 1`] = `
...
             <div class=\\"top\\">
                mr_mounty
             </div>
-      <div class=\\"bottom\\">
-         cosmosvaladdr…8yqw
+      <div class=\\"short-address\\">
+         <div id=\\"address\\" class=\\"address\\">
+            cosmosvalpub1234
+         </div>
+      <div class=\\"success\\">
```
Comment (faboweb) [inline test/unit/specs/components/staking/__snapshots__/TabValidators.spec.js.snap]: It is a little strange to have this DOM everywhere. How about using `v-if` instead of an `.active` class?

File: test/unit/specs/components/wallet/PageWallet.spec.js [hunk 11]
```diff
@@ -6,7 +6,10 @@ describe(`PageWallet`, () => {
 
   beforeEach(async () => {
     let instance = mount(PageWallet, {
-      stubs: { "modal-search": `<modal-search />` }
+      stubs: {
+        "modal-search": `<modal-search />`,
+        "short-address": `<short-address />`
```
Comment (faboweb) [inline test/unit/specs/components/wallet/PageWallet.spec.js]: this would also hide api incompatibility

File: app/src/renderer/components/common/ShortAddress.vue (hunk 5 above)
Comment (jbibla) [reply app/src/renderer/components/common/ShortAddress.vue]: not sure. i copied this code from `scripts/common`.

File: app/src/renderer/components/wallet/PageWallet.vue (hunk 6 above)
Comment (jbibla) [reply app/src/renderer/components/wallet/PageWallet.vue]: the ux for copying form inputs was unsatisfactory. we can discuss ways of improving this.

the short address component is on every page and includes functionality for copying to clipboard (and the tooltip). no more copying to clipboard triggering notify.

File: test/unit/specs/components/common/ShortAddress.spec.js (hunk 7 above)
Comment (fedekunze) [reply test/unit/specs/components/common/ShortAddress.spec.js]: why `cosm…asdf` ? shouldn't it show the whole prefix + the last 4 chars ?

Comment (faboweb) [reply test/unit/specs/components/common/ShortAddress.spec.js]: I created a follow up: https://github.com/cosmos/voyager/issues/1437

File: test/unit/specs/components/common/TmBalance.spec.js (hunk 9 above)
Comment (faboweb) [reply test/unit/specs/components/common/TmBalance.spec.js]: Example: If you use a property `address` for `ShortAdress` in `TmBalance` but the actually property used in `ShortAdress` is `bech32` the `TmBalance` test will pass even though the actual application will fail.

File: app/src/renderer/components/common/ShortBech32.vue [hunk 12]
```diff
@@ -1,7 +1,7 @@
 <template lang="pug">
 .short-bech32
   .address(@click.prevent.stop="copy" v-tooltip.top="address")#address {{ shortBech32 }}
-  .success(:class="{active:showSuccess}")
+  .copied(:class="{active:showSuccess}")
```
Comment (faboweb) [inline app/src/renderer/components/common/ShortBech32.vue]: my comment wasn't about the class but about the DOM element being there all the time. did you check if this has sth. todo with the animation?

Comment (jbibla) [reply app/src/renderer/components/common/ShortBech32.vue]: ah i misunderstood. still, i think this is better. 

it _does_ have to do with the animation. currently, the element is `position: relative` and so if you use a `v-if` then it will bump things around it. also, transitions work when changing css properties, not inserting new elements into the DOM.

it could be changed to allow for a `v-if` but i'm not sure that would be _better_.

Comment (faboweb) [reply app/src/renderer/components/common/ShortBech32.vue]: hm ok
